### ISSUE TITLE
[NEED] Code quality and Security

### ISSUE THREAD

**Issue by JFQueralt** [TheIOFoundation/ProjectLockdown#37]
**Problem**
Maintaining a large, community oriented codebase is always prone to potential errors and security problems.

**Objectives**
To identify one or more tools that can supervise Project Lockdown's codebase and trigger warnings

**Requirements**
- Must have
Free / Community / Open Source / Nonprofit free license.

Codebase review
-- Quality
-- Security

- Should have
Ability to generate Issues in GitHub.
Web based (serverless for Project Lockdown)

- Could have
Server based (we would then resort to spin our instance somewhere - Cost to be evaluated)

- Won't have

**Extra information**
Similar products available
- Sonarqube: https://www.sonarqube.org/

*REQUIRED* update wiki: https://github.com/TheIOFoundation/ProjectLockdown/wiki/Dev-Guidelines

**Pending**

**Resources**

**Related Issues**

**Comment by jeff-knurek** [comment]
here's a good list of some of the existing tools for static analysis of JS code:
https://blog.logrocket.com/static-analysis-in-javascript-11-tools-to-help-you-catch-errors-before-users-do/
**tldr** Prettier & ESLint are the top choices.

Alternatively, it looks like https://sonarcloud.io/ will be a good. At least the visualization looks nice, and it's based off a subset of eslint.

**Comment by JFQueralt** [comment]
SonarCloud looks like an interesting option.
Let me know if you want me to proceed with registering.

![image.png](https://images.zenhubusercontent.com/5f9c558d6913095f676e39b9/91124624-5df2-48a3-a7dc-b21a53c86a53)

**Comment by jeff-knurek** [comment]
Dependabot -> want to be able to create issues immediately - and assign to the appropriate team 

(it auto opens PRs) => who should review/merge?

**Comment by filipkowal** [comment]
If it comes to Sonar rules being checked while typing, there is this VSCode (and other IDE's) [extension](https://sonarcloud.io/sonarlint) that does it. We don't need to configure eslint to do it.
IMHO, we can install SonarCloud and inform everyone to install SonarLint on their IDE's.

**Comment by jeff-knurek** [comment]
@JFQueralt I think we're ready to enable SonarCloud, maybe you can do the honors in case there's extra registration steps :pray:

**Comment by JFQueralt** [comment]
Done.

Please check that things are working as expected. From what I understand, anyone in the Repo should be able to access the organization/project account:

https://sonarcloud.io/dashboard?id=TheIOFoundation_ProjectLockdown
(Will add it to Basaas in a moment)


See screenshot:
![image](https://user-images.githubusercontent.com/9198668/108221967-4b186600-7173-11eb-8189-c025827e5069.png)
![image](https://user-images.githubusercontent.com/9198668/108222000-54a1ce00-7173-11eb-9b3e-69302e9e32e3.png)

**Comment by JFQueralt** [comment]
Added [PLD] SonarCloud app on Basaas (Devs Group).

**Comment by jeff-knurek** [comment]
great :tada: 
I have access, and have seen the quality comments in the open PRs already, which so far look good - we may need to tweak the configuration as time goes by, but it's a good start
